core/src/wsv.rs

```diff
@@ -124,12 +125,54 @@ impl<W: WorldTrait> WorldStateView<W> {
         }
     }
 
+    #[cfg(feature = "roles")]
+    /// Returns a set of permission tokens granted to this account as part of roles and separately.
+    pub fn account_permission_tokens(
+        &self,
+        account: &Account,
+    ) -> iroha_data_model::account::Permissions {
+        let mut tokens = account.permission_tokens.clone();
+        for role_id in &account.roles {
+            if let Some(role) = self.world.roles.get(role_id) {
+                let mut role_tokens = role.permissions.clone();
+                tokens.append(&mut role_tokens);
+            }
+        }
+        tokens
+    }
+
+    #[allow(clippy::unused_self)]
+    #[cfg(not(feature = "roles"))]
+    /// Returns a set of permission tokens granted to this account as part of roles and separately.
+    pub fn account_permission_tokens(
+        &self,
+        account: &Account,
+    ) -> iroha_data_model::account::Permissions {
+        account.permission_tokens.clone()
+    }
+
+    /// Apply instructions to the `WorldStateView<W>`.
+    ///
+    /// # Errors
+    /// Can fail if execution of instructions fail(should be fine after validation)
+    fn proceed(&self, tx: &VersionedValidTransaction) -> Result<()> {
+        let tx = tx.as_inner_v1();
+
+        for instruction in &tx.payload.instructions {
+            instruction
+                .clone()
+                .execute(tx.payload.account_id.clone(), self)?;
+        }
+        // XXX: Should it just return `()`?
```

Comment:
And panic? There is a potential error here, can we guarantee that it never gets raised. You could remove the `OK(())`, though.